EDU-875: Update support info on Cloud pricing page (#2148)
temporalio/documentation: docs-src/cloud-context/pricing.md

With a subscription to Temporal Cloud, you also gain access to our support organization of developer success engineers and solution architects. Our experts assist with a range of work streams from Workflow design reviews to setting up observability to break/fix support with an agreed-upon set of SLAs.

The members of the Temporal Developer Success team are engineers who are Temporal experts and open source contributors. They're backed by the main contributors to the open source Temporal project. They provide deep knowledge of how Temporal works and how you can optimize your deployment.

They also investigate your Workflows to optimize their environments and possibly reduce costs associated with Actions and storage. They ensure your instance is performance tuned and help with other ongoing maintenance, like upgrades of the Temporal software and maintenance of the platform.

> If an issue occurs, the team provides support through our [portal](/cloud/how-to-create-a-ticket-for-temporal-support), [community forum](https://community.temporal.io/), knowledge base, and an optional dedicated Slack channel. Temporal offers two levels of support defined by their availability and SLAs.

Reviewer: ```suggestion
If an issue occurs, the team provides support through our [support portal](/cloud/how-to-create-a-ticket-for-temporal-support), [community forum](https://community.temporal.io/), and with Premium support, a dedicated Slack channel. Temporal offers two levels of support defined by their availability and SLAs.
```

- "portal" by itself doesn't make sense to me
- don't know what proving support through our knowledge base means, unless it's the existence of the docs KB, in which case link to it
- channel is for premium ([source](https://temporaltechnologies.slack.com/archives/C025MMBNTEV/p1687559261321139?thread_ts=1687557355.513249&cid=C025MMBNTEV))
